### ISSUE TITLE
Fix support of non-GET (POST, PUT, ...) requests rewriting

### ISSUE THREAD

**Issue by benoit74** [openzim/warc2zim#308]
Currently, non-GET (POST, PUT, ...) requests returning an HTML document are supposed to work but they are not tested at all. 

It is supposed to work based on what has been transferred from wabac.js, but not tested.

We need to :
- create a real test case on test website
- ensure rewrite works as expected on these requests
- include this non-GET requests dataset in the automated tests

**Comment by rgaudin** [comment]
I'd like to stress that it sorta work by accident because most readers simply ignore the HTTP method of the request and return content from the ZIM using the request's path.

At the moment, it is not harmonized. For instance apple reader ignore method so even DELETE would work but kiwix-serve (libkiwix) only replies to GET, HEAD and POST (so PUT doesn't work there).

I advocated for reader implementation guidelines but this was considered superfluous so I think this ticket will need to map the need and create upstream tickets on all concerned readers.

**Comment by benoit74** [comment]
I strongly agree. We should probably wait for a real usecase which would help make the issue way more understandable / less virtual.

Maybe we should even split this issue, first considering only POST, HEAD and PUT (most realistic to be supported) and then DELETE (not sure how we could support this).

**Comment by rgaudin** [comment]
I agree we should focus on use cases but for the sake of the argument, I don't see any difference between a POST and a DELETE. In both case, the source website probably managed data and returned something (even a 204) and as far as warc2zim is concerned, it's a request and a response. The fact that we loos any dynamic stuff is already acted even for GET with querystrings
